Backpack/OverlayView/Classes/BPKOverlayView.m

```diff
@@ -0,0 +1,192 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKOverlayView.h"
+
+#import <Backpack/Color.h>
+#import <Backpack/Common.h>
+#import <Backpack/Duration.h>
+#import <Backpack/Radii.h>
+#import <Backpack/Shadow.h>
+#import <Backpack/Spacing.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BPKOverlayView ()
+
+@property(nonatomic, strong, nullable) CALayer *tintLayer;
+
+@end
+
+@implementation BPKOverlayView
+
+@synthesize foregroundView = _foregroundView;
+@synthesize backgroundView = _backgroundView;
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    BPKAssertMainThread();
+    self = [super initWithCoder:aDecoder];
+
+    if (self) {
+        [self setupWithOverlayType:BPKOverlayViewOverlayTypeTint cornerStyle:BPKOverlayViewCornerStyleNone];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    BPKAssertMainThread();
+    self = [super initWithFrame:frame];
+
+    if (self) {
+        [self setupWithOverlayType:BPKOverlayViewOverlayTypeTint cornerStyle:BPKOverlayViewCornerStyleNone];
+    }
+
+    return self;
+}
+
+- (nullable instancetype)initWithOverlayType:(BPKOverlayViewOverlayType)overlayType cornerStyle:(BPKOverlayViewCornerStyle)cornerStyle {
+    BPKAssertMainThread();
+    self = [super initWithFrame:CGRectZero];
+
+    if (self) {
+        [self setupWithOverlayType:overlayType cornerStyle:cornerStyle];
+    }
+
+    return self;
+}
+
+- (void)layoutSubviews {
+    if (self.tintLayer == nil) {
+        return;
+    }
+
+    self.tintLayer.frame = self.bounds;
+}
+
+#pragma mark - Private
+
+- (void)setupWithOverlayType:(BPKOverlayViewOverlayType)overlayType cornerStyle:(BPKOverlayViewCornerStyle)cornerStyle {
+    self.overlayType = overlayType;
+    self.cornerStyle = cornerStyle;
+}
+
+#pragma mark - getters
+
+- (UIView *)foregroundView {
+    if (_foregroundView != nil) {
+        return _foregroundView;
+    }
+
+    _foregroundView = [[UIView alloc] initWithFrame:CGRectZero];
+    _foregroundView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self addSubview:_foregroundView];
+
+    [NSLayoutConstraint activateConstraints:@[
+        [_foregroundView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+        [_foregroundView.topAnchor constraintEqualToAnchor:self.topAnchor],
+        [self.trailingAnchor constraintEqualToAnchor:_foregroundView.trailingAnchor],
+        [self.bottomAnchor constraintEqualToAnchor:_foregroundView.bottomAnchor]
+    ]];
+
+    return _foregroundView;
+}
+
+- (UIView *)backgroundView {
+    if (_backgroundView != nil) {
+        return _backgroundView;
+    }
+
+    self.tintLayer = [CALayer layer];
+    self.tintLayer.backgroundColor = BPKColor.skyBlueShade03.CGColor;
+    self.tintLayer.zPosition = 100;
```

Comment:
This feels arbitrary and magical 🤔

Reply:
Changed to `1`. It seems that it's still needed sadly